spell/psm-checklists.md

```diff
@@ -0,0 +1,113 @@
+# PSM Checklists
+
+## PSM Migration Checklist
+
+- [ ] IF contracts are being on-boarded in the current spell, execute the full [LitePSM On-boarding Checklist](#litepsm-on-boarding-checklist)
+- [ ] IF a PSM is deprecated in the migration, execute the full [PSM Off-boarding Checklist](#psm-off-boarding-checklist)
+- [ ] Amount of `gem` being migrated matches the Exec Sheet
+- [ ] Collateral amount that is fetched from the source PSM matches the collateral is added to the destination PSM
+- [ ] No bad debt is left behind
+- Source PSM:
+    - [ ] ilk matches source PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+- Destination PSM:
+    - [ ] ilk matches destination PSM
+    - [ ] IF present, `tin` matches the value in the Exec Sheet
+    - [ ] IF present, `tout` matches the value in the Exec Sheet
+    - [ ] `AutoLine` is updated according to the Exec Sheet
+    - [ ] `buf` matches the value in the Exec Sheet
+    - [ ] `fill` is called to ensure liquidity is immediately available after spell execution
+
+## LitePSM On-boarding Checklist
+
+- Deployed Contracts
+    - `DssLitePsm`
+        - [ ] Uses the agreed best Solidity compiler version at the time
+        - [ ] Optimizer is **enabled**
+        - [ ] Contract has been reviewed by at least a couple of DAO dev teams
+        - [ ] Contract has been audited by an external party
+            - [ ] Audit reports are public
```

Comment:
Does it mean externally verifiable or should I trust PDF uploaded in the repo? 🙂

Reply:
Yeah, there's not an easy way to remove trust from this process.
ChainSecurity publishes all audit reports in their [website](https://www.chainsecurity.com/smart-contract-audit-reports?client=MakerDao&topic=Stablecoins), so the reviewer could fetch it from there. However, in this specific case, the audit report that is [published](https://www.chainsecurity.com/security-audit/makerdao-psm-lite) is outdated &ndash; the latest version is 11, while the link points to version 8.

Reply:
I think trusting PDFs uploaded by the same person who wrote the code is not a good idea, as in case of malicious intent that would equal of having no audit / planted vulnerability.

That's why I would propose that this check more specifically requires finding external verification of the audit. This can be either: a) PDF found in the auditors website (technically it's their responsibility to update PDF to the latest version) b) signed /attested PDF by externally verifiable auditors public key c) contacting auditors and asking them to update either of the above.

```suggestion
            - [ ] Audit report source could be confirmed (e.g. found on their website)
```

Reply:
I'm asking ChainSecurity to update their website with the latest version, because it would be unfeasible to start doing attestations by the time the next spell is due.
Happy do discuss how to include that into the workflow for newer modules moving forwards.